src/Common/src/Interop/Unix/libcurl/Interop.libcurl_types.cs

```diff
@@ -29,5 +54,30 @@ internal static partial class CURLcode
         {
             internal const int CURLE_OK = 0;
         }
+
+        // Class for constants defined for the enum curlioerr in curl.h
+        internal static partial class curlioerr
+        {
+            internal const int CURLIOE_OK = 0;
+            internal const int CURLIOE_UNKNOWNCMD = 1;
+            internal const int CURLIOE_FAILRESTART = 2;
+        }
+
+        // Class for constants defined for the enum curliocmd in curl.h
+        internal static partial class curliocmd
+        {
+            internal const int CURLIOCMD_RESTARTREAD = 1;
+        }
+
+        public delegate ulong curl_readwrite_callback(
+            IntPtr buffer,
+            ulong size,
+            ulong nitems,
```

Comment:
In the callback logic, we need to calculate the total size of the data by multiplying size and nitems. If I define these as IntPtr (size_t), will it matter that I would have to cast to ulong?